include/nuttx/semaphore.h

```diff
@@ -184,9 +184,17 @@ int nxsem_destroy(FAR sem_t *sem);
  *
  ****************************************************************************/
 
-int nxsem_wait(FAR sem_t *sem);
 int nxsem_wait_slow(FAR sem_t *sem);
 
+#if !defined(CONFIG_SMP) && defined(CONFIG_BUILD_FLAT)
+static inline int nxsem_wait(FAR sem_t *sem)
```

Comment:
do you make some benchmark? The fast path is still better even in flat/up case from our test from our test.

Reply:
Doesn't this just depend on how contested the semaphores are ?

For code size, it depends if the chip can do atomic_try_cmpxchg_acquire without locking with a single atomic instruction. The locking adds extra **inline** code, this is why the code size increases so dramatically for stm32f3 and in this case, taking the slow path directly should also be faster.

I'm wondering if it's possible to use __atomic_always_lock_free somehow to enable/disable the fast paths? If not, then the fast path must be possible to disable in some other way.

Reply:
I can leave this one out as well from this PR, it was just to save some hundred bytes of flash...

Reply:
> Doesn't this just depend on how contested the semaphores are ?
> 

the fast path has the effect only when the contest doesn't exist, the performance should be a little bit slow for contested case with the new change.